main.go

```diff
@@ -91,17 +104,33 @@ func getModules() ([]*modEntry, error) {
 
 	for _, mod := range mods {
 		rev := mod.Version
+		url, err := vcs.RepoRootForImportPath(mod.Path, false)
+		if err != nil {
+			return nil, err
+		}
+
 		if commitShaRev.MatchString(rev) {
 			rev = commitShaRev.FindAllStringSubmatch(rev, -1)[0][1]
 		} else if commitRevV2.MatchString(rev) {
 			rev = commitRevV2.FindAllStringSubmatch(rev, -1)[0][1]
 		} else if commitRevV3.MatchString(rev) {
 			rev = commitRevV3.FindAllStringSubmatch(rev, -1)[0][1]
+		} else if mod.Path != url.Root {
+			subPath := strings.Split(mod.Path, url.Root+"/")[1]
+			if !versionNumber.MatchString(subPath) {
```

Comment:
The `subPath` might be like `v2/modulePath`, in this case the rev should be prepended with `modulePath`?
Also, if the `subPath` is `v2modulePath`, the rev should also be prepended with `v2modulePath`?

Reply:
Can the v2 be before the module path? I thought it can only be at the very end. Do you have a real world example?